spharpy/classes/audio.py

```diff
@@ -0,0 +1,232 @@
+
+        # set normalization
+        if normalization not in ["n3d", "maxN", "sn3d"]:
+            raise ValueError("Invalid normalization, has to be 'sn3d', "
+                             f"'n3d', or 'maxN', but is {normalization}")
+        self._normalization = normalization
+
+        # set channel_convention
+        if channel_convention not in ["acn", "fuma"]:
+            raise ValueError("Invalid channel convention, has to be 'acn' "
+                             f"or 'fuma', but is {channel_convention}")
+        self._channel_convention = channel_convention
+
+        # set condon shortley
+        if not isinstance(condon_shortley, bool) and condon_shortley != 'auto':
+            raise ValueError(
+                "Condon_shortley has to be a bool, or 'auto'.")
+
+        if condon_shortley == 'auto' and basis_type == 'complex':
+            self._condon_shortley = True
+        elif condon_shortley == 'auto' and basis_type == 'real':
+            self._condon_shortley = False
+
+        Signal.__init__(self, data, sampling_rate=sampling_rate,
+                        n_samples=n_samples, domain=domain, fft_norm=fft_norm,
+                        comment=comment, is_complex=is_complex)
+
+    @property
+    def n_max(self):
+        """the maximum Spherical Harmonics order."""
+        return self._n_max
+
+    @property
+    def basis_type(self):
+        """the type of the Spherical Harmonics bases (complex or real)."""
+        return self._basis_type
+
+    @property
+    def normalization(self):
+        """the normalization of the Spherical Harmonics coefficients."""
+        return self._normalization
+
+    @normalization.setter
+    def normalization(self, value):
+        """set the normalization of the Spherical Harmonics coefficients."""
+        if self.normalization is not value:
+            self._renormalize(value)
+
+    @property
+    def condon_shortley(self):
+        return self._condon_shortley
+
+    @property
+    def channel_convention(self):
+        """the channel convention of the Spherical Harmonics coefficients."""
+        return self._channel_convention
+
+    @channel_convention.setter
+    def channel_convention(self, value):
+        """
+        set the channel convention of the Spherical Harmonics coefficients.
+        """
+        if value not in ["acn", "fuma"]:
+            raise ValueError("Invalid channel convention, has to be 'acn' "
+                             f"or 'fuma', but is {value}")
+
+        if self.channel_convention is not value:
+            self._change_channel_convention()
+
+    def _renormalize(self, value):
+        if value not in ["n3d", "maxN", "sn3d"]:
+            raise ValueError("Invalid normalization, has to be 'sn3d', "
+                             f"'n3d', or 'maxN', but is {value}")
+        acn = np.arange((self.n_max + 1) ** 2)
+
+        if self.channel_convention == "fuma":
+            orders, _ = fuma_to_nm(acn)
```

Comment:
```suggestion
            orders = fuma_to_nm(acn)[0]
```
Writing to _ is not really advised :)

But I guess these comments should now go to the new branch where these methods are implemented as functions